src/lightning_app/cli/commands/connection.py

```diff
@@ -27,13 +28,16 @@ def connect(app_name_or_id: str, yes: bool = False):
     """Connect to a Lightning App."""
     from lightning_app.utilities.commands.base import _download_command
 
-    _clean_lightning_connection()
+    if _CLEAN_LIGHTNING_CONNECTION:
+        _clean_lightning_connection()
```

Comment:
I see `_clean_lightning_connection` is called every time when [executing a command](https://github.com/Lightning-AI/lightning/blob/921dc1cd9afbdbf33589d8b45f390c7957072f93/src/lightning_app/cli/commands/app_commands.py#L24), and since it has a new pid it looks like it will delete it.

Reply:
No because this is going to be the pid of the terminal process, not the pid of the command itself.